docs/user_manual/components.md

```diff
@@ -721,6 +721,110 @@ $$
+| `i_angle_sigma`          | `double`                                                                      | rad        | standard deviation of the current (`i`) phase angle measurement error. Usually this is the absolute measurement error range divided by 3. | &#10060; | &#10004; |                        `> 0`                         |
+
+#### Current Sensor Concrete Types
+
+```{warning}
+At the time of writing, this feature is still experimental and is not yet publicly available.
+```
+
+There are two concrete types of current sensor. They share similar attributes:
+the meaning of `RealValueInput` is different, as shown in the table below.
+
+| type name             | meaning of `RealValueInput` |
+| --------------------- | --------------------------- |
+| `sym_current_sensor`  | `double`                    |
+| `asym_current_sensor` | `double[3]`                 |
+
+##### Input
+
+| name               | data type        | unit       | description                               |              required              |  update  |
+| ------------------ | ---------------- | ---------- | ----------------------------------------- | :--------------------------------: | :------: |
+| `i_measured`       | `RealValueInput` | ampere (A) | measured current (`i`) magnitude          | &#10024; only for state estimation | &#10004; |
+| `i_angle_measured` | `RealValueInput` | rad        | measured phase angle of the current (`i`) | &#10024; only for state estimation | &#10004; |
+
+See the documentation on [state estimation calculation methods](calculations.md#state-estimation-algorithms) for details per method on how the variances are taken into account for both the global and local angle measurement types and for the individual phases.
+
+##### Steady state output
+
+```{note}
+A sensor only has output for state estimation. For other calculation types, sensor output is undefined.
+```
+
+| name               | data type         | unit       | description                                                                                 |
+| ------------------ | ----------------- | ---------- | ------------------------------------------------------------------------------------------- |
+| `i_residual`       | `RealValueOutput` | ampere (A) | residual value between measured current (`i`) and calculated current (`i`)                  |
+| `i_angle_residual` | `RealValueOutput` | rad        | residual value between measured phase angle and calculated phase angle of the current (`i`) |
+
+#### Electric Model
+
+`Generic Current Sensor` is modeled by following equations:
+
+$$
+   \begin{eqnarray}
+        & p_{\text{residual}} = p_{\text{measured}} - p_{\text{state}} \\
+        & q_{\text{residual}} = q_{\text{measured}} - q_{\text{state}}
+   \end{eqnarray}
+$$
+
+Internally, the current phasor measurement is decomposed into a separate real and imaginary component
```

Comment:
I feel the variance explanation needs to be moved to calculations under the state estimation heading. The workout is too complex for this page.